embassy-nrf/src/usb.rs

```diff
@@ -595,29 +619,19 @@ impl<'d, T: Instance> driver::ControlPipe for ControlPipe<'d, T> {
             let req = self.request.unwrap();
             assert_eq!(req.direction, UsbDirection::Out);
             assert!(req.length > 0);
-            assert!(buf.len() >= usize::from(req.length));
-
-            let regs = T::regs();
 
-            // Wait until ready
-            regs.intenset.write(|w| w.ep0datadone().set());
-            poll_fn(|cx| {
-                EP_OUT_WAKERS[0].register(cx.waker());
-                let regs = T::regs();
-                if regs
-                    .events_ep0datadone
-                    .read()
-                    .events_ep0datadone()
-                    .bit_is_set()
-                {
-                    Poll::Ready(())
-                } else {
-                    Poll::Pending
+            let req_length = usize::from(req.length);
+            let max_packet_size = usize::from(self.max_packet_size);
+            let mut total = 0;
+            for chunk in buf.chunks_mut(max_packet_size) {
+                let size = self.read(chunk).await?;
+                total += size;
+                if size < max_packet_size || total == req_length {
```

Comment:
Is `total == req_length` needed? I think chunked transfers are always required to end in a packet smaller than `max_packet_size`, inserting a ZLP if it's an exact multiple.

Reply:
Oh huh, it's [not actually required](https://stackoverflow.com/questions/3739901/when-do-usb-hosts-require-a-zero-length-in-packet-at-the-end-of-a-control-read-t).